### PR TITLE
Introducing Rubberduck.Resources

### PR DESCRIPTION
**Pull request by retailcoder**
This will likely conflict with #3975 a bit.

ref. #3862

### CODE REVIEW

**Comment by bclothier** [comment]
I do think this is a case of choosing your own poisons. That said, I can offer one reason to centralize the resources --- it eliminate the guesswork of "do I need to share it?". A changing of resource is now only a namespace realignment; no project references needs to be changed, and though it's not strictly necessary, there is something to be said for having all i18n stuff in one place. 

The bad is that of course, it can turn into a kitchen sink and we so totally don't want it. Also when we do change from `Rubberduck.Core` to `Rubberduck.UI`, `Rubberduck.Resources` could seem like hair splitting because 90%+ of resources are for UI.

So yeah, pick one or other and be happy/sad.

**Comment by retailcoder** [comment]
@bclothier okay then I'll pick the "let's put everything we can under `Rubberduck.Resources`"; the DLL itself might not need to be updated anytime soon, but if we add Italian and Spanish translations then we should be able to do that without needing to recompile the regex assistant.

Thoughts?

File: Rubberduck.Core/Settings/HotkeySetting.cs [hunk 1]
```diff
@@ -29,14 +29,7 @@ public class HotkeySetting
...
 
-        public override string ToString()
-        {
-            return string.Format("{0}{1}{2}{3}",
-                HasCtrlModifier ? KeyModifierCtrl : string.Empty,
-                HasShiftModifier ? KeyModifierShift : string.Empty,
-                HasAltModifier ? KeyModifierAlt : string.Empty,
-                Key1);
-        }
+        public override string ToString() => $"{(HasCtrlModifier ? KeyModifierCtrl : string.Empty)}{(HasShiftModifier ? KeyModifierShift : string.Empty)}{(HasAltModifier ? KeyModifierAlt : string.Empty)}{Key1}";
```
Comment (bclothier) [inline Rubberduck.Core/Settings/HotkeySetting.cs]: what's up with this? seems less readable?

File: Rubberduck.Main/Rubberduck.Main.csproj [hunk 2]
```diff
@@ -141,7 +141,8 @@
     <Optimize>false</Optimize>
   </PropertyGroup>
   <PropertyGroup>
-    <ApplicationIcon>Ducky.ico</ApplicationIcon>
+    <ApplicationIcon>
+    </ApplicationIcon>
```
Comment (bclothier) [inline Rubberduck.Main/Rubberduck.Main.csproj]: Is that intentional?

File: Rubberduck.VBEditor.VBA/Rubberduck.VBEditor.VBA.csproj [hunk 3]
```diff
@@ -112,7 +112,7 @@
       <EmbedInteropTypes>False</EmbedInteropTypes>
     </Reference>
     <Reference Include="stdole, Version=7.0.3300.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <EmbedInteropTypes>True</EmbedInteropTypes>
+      <EmbedInteropTypes>False</EmbedInteropTypes>
```
Comment (bclothier) [inline Rubberduck.VBEditor.VBA/Rubberduck.VBEditor.VBA.csproj]: Do we want that? I thought we should be embedding to avoid problems w/ versioning?

Comment (retailcoder) [reply Rubberduck.VBEditor.VBA/Rubberduck.VBEditor.VBA.csproj]: Hmm I toggled that in response to a build warning.

File: Rubberduck.Main/Rubberduck.Main.csproj (hunk 2 above)
Comment (retailcoder) [reply Rubberduck.Main/Rubberduck.Main.csproj]: Nope, WIP :smiley:

File: Rubberduck.Core/Settings/HotkeySetting.cs (hunk 1 above)
Comment (retailcoder) [reply Rubberduck.Core/Settings/HotkeySetting.cs]: Yeah, agreed. will revert.

Comment (retailcoder) [reply Rubberduck.Core/Settings/HotkeySetting.cs]: Done.

**Comment by retailcoder** [comment]
This PR is pretty much impossible to review, but these two screenshots summarize pretty much everything that's happening (`Meta.resx` was renamed to `InspectionInfo.resx`):

image

image

Some cleanup still needs to happen with `Resources.resx` (contains a number of icons and images) and `RubberduckUI.resx` needs to be sliced.

"About" resource keys in `RubberduckUI` have already been extracted into `AboutUI.resx`, but I'll stop it here before hitting 1000 affected files, and continue the slicing of `RubberduckUI.resx` in another PR - I'll make one PR per feature group, to limit the scope.

Thoughs on whether it's a good idea to merge this now and continue later, or if more changes should be included here?

**Comment by bclothier** [comment]
My vote would be to merge it and try to keep subsequent reorganization PR limited in the scope. It looks better on the history and helps avoid hiding any potential mistakes.

**Pull request merged**